ui/src/App.tsx

```diff
@@ -1,52 +1,81 @@
 //External components
 import { ThemeProvider } from "@mui/material/styles";
-import { Container } from "@mui/material";
+import { Container, Alert } from "@mui/material";
 
 //Internal components
 import TopBar from "./components/TopBar/TopBar";
 import ImportScreen from "./ImportScreen";
 import ValidatorList from "./components/ValidatorList/ValidatorList";
-import Message from "./components/Messages/Message";
-
-//Styles
-import "./App.css";
+import { consensusClient, executionClient } from "./params";
 
 //Themes
 import { darkTheme } from "./Themes/globalThemes";
 
 //Logic
-import { getUrlParams } from "./getUrlParams";
+import { getUrlParams } from "./logic/Utils/getUrlParams";
 import { Web3SignerApi } from "./apis/web3signerApi";
 
 //Other libraries
 import { BrowserRouter, Route, Routes } from "react-router-dom";
 import React, { useEffect } from "react";
+import ClientsBox from "./components/ClientsBox/ClientsBox";
 
 function App() {
   const [network, setNetwork] = React.useState("");
   const [web3signerApi, setWeb3signerApi] =
     React.useState<Web3SignerApi | null>(null);
+  const [signerStatus, setSignerStatus] = React.useState("loading");
 
   useEffect(() => {
-    const { network, authToken, host, baseUrl } = getUrlParams();
+    const { network, authToken, host, signerUrl } = getUrlParams();
     setNetwork(network);
-    if (baseUrl) setWeb3signerApi(new Web3SignerApi(baseUrl, authToken, host));
+    if (signerUrl) {
+      setWeb3signerApi(
+        new Web3SignerApi({
+          baseUrl: signerUrl,
+          authToken: authToken,
+          host: host,
+        })
+      );
+    }
   }, []);
 
+  //Status check for web3signer
+  const showSignerStatus = async () => {
+    setSignerStatus((await web3signerApi?.getStatus())?.status || "LOADING");
+  };
+
+  useEffect(() => {
+    showSignerStatus();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [web3signerApi]);
+
   return (
     <ThemeProvider theme={darkTheme}>
-      <TopBar network={network} />
+      <TopBar network={network} signerStatus={signerStatus} />
       <Container component="main" maxWidth="xl">
-        {web3signerApi ? (
+        {web3signerApi && network ? (
```

Comment:
You should check the status of the web3signerApi before moving forward, it may be "error"

Reply:
Done. https://github.com/dappnode/eth2-keymanager-frontend/commit/08835d4d1ff97e9596e327e8de6126a570181eb4